### PR TITLE
Enable Codec 2 Auto EQ for 800XA

### PR DESCRIPTION
**Pull request by drowe67**
Forgot one FreeDV mode in https://github.com/drowe67/codec2/pull/194 ! 800XA also uses codec 2 700C ....

### CODE REVIEW

File: src/freedv_api.c [hunk 1]
```diff
@@ -1250,7 +1250,8 @@ void freedv_set_phase_est_bandwidth_mode(struct freedv *f, int val) {
 }
 
 void freedv_set_eq(struct freedv *f, int val) {
-    if (FDV_MODE_ACTIVE( FREEDV_MODE_700C, f->mode) || FDV_MODE_ACTIVE( FREEDV_MODE_700D, f->mode) || FDV_MODE_ACTIVE( FREEDV_MODE_700E, f->mode)) {
+    if (FDV_MODE_ACTIVE( FREEDV_MODE_700C, f->mode) || FDV_MODE_ACTIVE( FREEDV_MODE_700D, f->mode) ||
```
Comment (tmiw) [inline src/freedv_api.c:1253]: Maybe checks like these can be consolidated in the individual Codec2 modes? For instance, `if (codec2_is_mode_700c(f->mode)) { ... }`.

Comment (drowe67) [reply src/freedv_api.c:1253]: :+1:  Yes I was thinking along similar lines, e.g. if we add a field to note the codec 2 mode to the freedv states:
```
if (f->codec2_mode == CODEC2_MODE_700C)
```

**Comment by drowe67** [comment]
@tmiw - I poked around a bit and found a simpler solution, the `codec2_700c_eq()` function is only active for Codec 2 700C, so it's safe to call from other modes.  I think this is a reasonable design choice - "do nothing silently" if the requested API function is not relevant to the current mode.  Simplifies the upper layers.

Pls tell me what you think of this solution, and also:

https://github.com/drowe67/freedv-gui/pull/141

**Comment by tmiw** [comment]
@drowe67, looks good to me. 👍

**Pull request merged**